FrisyLULTest/Test/TestScreenshot.cs

```diff
@@ -0,0 +1,165 @@
+        /// </summary>
+        [TestMethod]
+        public void TestTakerWithInvalidProc()
+        {
+            Assert.IsTrue(Screenshot.TakeScreenshot(_invalidProcess).IsEmpty);
+        }
+
+        /// <summary>
+        /// Test what happens when user tries to screenshot a valid process.
+        /// A valid/ screenshot should not be empty.
+        /// </summary>
+        [TestMethod]
+        public void TestTakerWithValidProc()
+        {
+            Assert.IsFalse(Screenshot.TakeScreenshot(_validProcess).IsEmpty);
+        }
+
+        #endregion
+
+        #region Screenshot.Dispose
+
+        /// <summary>
+        /// Test the dispose method with a valid process/ screenshot.
+        /// The disposed object which is based on a valid process/ screenshot should
+        /// be throwing an exception when accessing its attributes.
+        /// </summary>
+        [TestMethod]
+        [ExpectedException(typeof(ArgumentException))]
+        public void TestDisposeWithValid()
+        {
+            var shot = Screenshot.TakeScreenshot(_validProcess);
+            shot.Dispose();
+
+            Assert.AreEqual(shot.Height, 0);
+        }
+
+        /// <summary>
+        /// Test the dispose method with an invalid process/ screenshot.
+        /// The disposed object which is based on an invalid process/ screenshot should
+        /// be height/ width = 0 (not optimally)
+        /// </summary>
+        [TestMethod]
+        public void TestDisposeWithInvalid()
+        {
+            var shot = Screenshot.TakeScreenshot(_invalidProcess);
+            shot.Dispose();
+
+            Assert.AreEqual(shot.Height + shot.Width, 0);
+        }
+
+        #endregion
+
+        #region Screenshot.Extract
+
+        /// <summary>
+        /// Test Extract on an empty screenshot.
+        /// An unvalid process/ screenshot should be empty.
+        /// </summary>
+        [TestMethod]
+        public void TestExtractWithEmpty()
+        {
+            Assert.IsTrue(Screenshot.TakeScreenshot(_notExistingProcess).Extract(0, 0, 2560, 1080).IsEmpty);
+        }
+
+        /// <summary>
+        /// Test Extract on an valid screenshot.
+        /// A valid process/ screenshot should not be empty.
+        /// </summary>
+        [TestMethod]
+        public void TestExtractWithValid()
+        {
+            Assert.IsFalse(Screenshot.TakeScreenshot(_validProcess).Extract(0, 0, 2560, 1080).IsEmpty);
+        }
+
+        #endregion
+
+        #region Screenshot.Save
+
+        /// <summary>
+        /// Test saving an empty screenshot.
+        /// An empty screenshot should have a height/ width of 1 pixel.
+        /// </summary>
+        [TestMethod]
+        public void TestSaveWithEmpty()
+        {
+            Screenshot.TakeScreenshot(_notExistingProcess).Save(@"tmp.png");
+            var img = Image.FromFile(@"tmp.png");
+
+            Assert.IsTrue(img.Size.Width == 1 && img.Size.Height == 1);
```

Comment:
save bei empty screenshot sollte am besten gar nix saven.  wenn paar tests fehlschlagen is gut dann kann ich das morgen im stream fixen